Lib/idlelib/calltips.py

```diff
@@ -139,17 +141,31 @@ def get_argspec(ob):
         ob_call = ob.__call__
     except BaseException:
         return argspec
-    if isinstance(ob, type):
-        fob = ob.__init__
-    elif isinstance(ob_call, types.MethodType):
+    if isinstance(ob_call, types.MethodType):
         fob = ob_call
     else:
         fob = ob
-    if isinstance(fob, (types.FunctionType, types.MethodType)):
-        argspec = inspect.formatargspec(*inspect.getfullargspec(fob))
-        if (isinstance(ob, (type, types.MethodType)) or
-                isinstance(ob_call, types.MethodType)):
-            argspec = _first_param.sub("", argspec)
+
+    try:
+        argspec = str(inspect.signature(fob))
+    except ValueError as err:
+        msg = str(err)
+        if msg.startswith(_invalid_method):
+            argspec = _invalid_method
+            return argspec
+        elif msg.startswith('no signature found for'):
+            """If no signature found for function or method"""
+            pass
+        else:
+            """Callable is not supported by signature"""
+            pass
```

Comment:
If we do this, it will break the unittest, for e.g. `range` will not able to return argspec.